src/components/Settings/SettingsPlex.tsx

```diff
@@ -108,14 +110,26 @@ const SettingsPlex: React.FC<SettingsPlexProps> = ({ onComplete }) => {
   const { addToast, removeToast } = useToasts();
   const PlexSettingsSchema = Yup.object().shape({
     hostname: Yup.string()
+      .nullable()
       .required(intl.formatMessage(messages.validationHostnameRequired))
       .matches(
         /^(([a-z]|\d|_|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])([a-z]|\d|-|\.|_|~|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])*)?([a-z]|\d|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])$/i,
         intl.formatMessage(messages.validationHostnameRequired)
       ),
     port: Yup.number()
-      .typeError(intl.formatMessage(messages.validationPortRequired))
+      .nullable()
       .required(intl.formatMessage(messages.validationPortRequired)),
+    webAppUrl: Yup.string()
+      .nullable()
+      .url(intl.formatMessage(messages.validationWebAppUrl))
+      .test(
+        'valid-web-app-url',
+        intl.formatMessage(messages.validationWebAppUrl),
+        (value) =>
+          !value ||
+          value === 'https://app.plex.tv/desktop' ||
+          value.endsWith('/web')
```

Comment:
I think I've seen people who have done some fancy Nginx rewrites to get rid of the `/web` or change it to another path.

Reply:
Oh hmm.  My concern is just that without a check, users who haven't done that might not enter a valid Plex Web URL, and that would result in none of the play buttons working. 🤔

Reply:
Also, there's nothing stopping those users who are doing fancy Nginx rewrites from modifying the value of this setting in their `settings.json` file.  If they're using special not-officially-supported configs to proxy their Plex install, it shouldn't be a problem for them to modify their settings file manually? 🤔

Reply:
This is sort of an advanced option anyways, no? Nothing is stopping someone from putting in a plex URL that doesn't work anyways. I don't think we should be so restrictive for a field thats entire purpose is to allow custom domains for their plex setup. And we should never design anything with the intent for people to have to manually edit settings.json to get it to work for them?

We can protect users but only to an extent. If they enable and use things they don't understand, we can't help them beyond that.

Reply:
That's true. I'll remove the validation and put the `Advanced` badge on this setting then?

Reply:
I think that's better. Maybe just a check that there is no trailing slash?

Reply:
I actually just verified that it still works with a trailing slash! (I tested with `https://plex.mydomain.com/web/`)

Reply:
Huh, well I guess a double slash in the URL wouldn't be the worst thing but it's technically incorrect.

Reply:
@sct It actually doesn't result in a double slash:
```ts
      this.plexUrl = `${
        webAppUrl ? webAppUrl : 'https://app.plex.tv/desktop'
      }#!/server/${machineId}/details?key=%2Flibrary%2Fmetadata%2F${
        this.ratingKey
      }`;
```

Reply:
Oh, right, they use a hash. Nevermind.